### ISSUE TITLE
Should be used sync function in when ? 

### ISSUE THREAD

**Issue by love8587** [frankthelen/rools#53]
I'm trying rules engine to implement evaludate articles to accept automatically.

I researched rules enginer and rools module looks good for me.

However, there is a part that has been hesitating. 

I think when conditions could be async functions. Is it should be compose of sync functions ?

**Comment by frankthelen** [comment]
yes, premises (when) are always working synchronously on the facts.

**Comment by love8587** [comment]
Thanks for your answer. Can I get a some refference to understand the reason why Promises are always working synchronously ?

**Comment by frankthelen** [comment]
Thanks for the question. The answer is probably for performance reasons and because I was thinking that it is not required. But why not. Let me think about it.
